### PR TITLE
Add matrix scoreboard endpoint

### LINKED ISSUE #10: Matrix Scoreboard Backend
Ref:
image

### PR DESCRIPTION
**Pull request by david-cooke**
Fixes #10

### CODE REVIEW

**Review by Bentechy66 — APPROVED** [review]

File: src/leaderboard/views.py [hunk 1]
```diff
@@ -74,3 +75,24 @@ def list(self, request, *args, **kwargs):
...
+        teams = Team.objects.filter(is_visible=True).order_by('-leaderboard_points', 'last_score').prefetch_related(
+            Prefetch('solves', queryset=Solve.objects.filter(correct=True))
+        )
+        for team in teams:
+            scoreboard.append({
+                'id': team.id,
+                'name': team.name,
+                'points': team.leaderboard_points,
+                'solves': list(team.solves.values_list('challenge', flat=True))
+            })
```
Comment (jerbob) [inline src/leaderboard/views.py:97]: Is there any `aggregate` magicks we could apply here to avoid iterating all entries and creating a dict from them?

Comment (david-cooke) [reply src/leaderboard/views.py:97]: I'd be surprised if there isn't

**Review by Bentechy66 — APPROVED** [review]
lgtm

**Review by jerbob — APPROVED** [review]

**Pull request merged**